File: Aerolinea/src/java/airline/model/Vuelo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package airline.model;

import java.util.Date;
import java.io.Serializable;

/**
 *
 * @author josealonso
 */
public class Vuelo implements Serializable, Jsonable{
    int numero_vuelo;
    Ciudad ciudad_origen;
    Ciudad ciudad_destino;
    boolean estado;
    int precio;
    String duracion;
    String hora;
    boolean oferta; // si el vuelo esta en oferta lo tira en el carrusel
    String imagen; //Direccion de la imagen
    String dia;
    int descuento;

    public Vuelo(int numero_vuelo, Ciudad ciudad_origen, Ciudad ciudad_destino, boolean estado, int precio, String duracion, String hora, boolean oferta, String imagen, String dia, int descuento) {
        this.numero_vuelo = numero_vuelo;
        this.ciudad_origen = ciudad_origen;
        this.ciudad_destino = ciudad_destino;
        this.estado = estado;
        this.precio = precio;
        this.duracion = duracion;
        this.hora = hora;
        this.oferta = oferta;
        this.imagen = imagen;
        this.dia = dia;
        this.descuento = descuento;
    }

    public Vuelo() {
    }

    public int getNumero_vuelo() {
        return numero_vuelo;
    }

    public void setNumero_vuelo(int numero_vuelo) {
        this.numero_vuelo = numero_vuelo;
    }

    public Ciudad getCiudad_origen() {
        return ciudad_origen;
    }

    public void setCiudad_origen(Ciudad ciudad_origen) {
        this.ciudad_origen = ciudad_origen;
    }

    public Ciudad getCiudad_destino() {
        return ciudad_destino;
    }

    public void setCiudad_destino(Ciudad ciudad_destino) {
        this.ciudad_destino = ciudad_destino;
    }

    public boolean isEstado() {
        return estado;
    }

    public void setEstado(boolean estado) {
        this.estado = estado;
    }

    public int getPrecio() {
        return precio;
    }

    public void setPrecio(int precio) {
        this.precio = precio;
    }

    public String getDuracion() {
        return duracion;
    }

    public void setDuracion(String duracion) {
        this.duracion = duracion;
    }

    public String getHora() {
        return hora;
    }

    public void setHora(String hora) {
        this.hora = hora;
    }

    public boolean isOferta() {
        return oferta;
    }

    public void setOferta(boolean oferta) {
        this.oferta = oferta;
    }

    public String getImagen() {
        return imagen;
    }

    public void setImagen(String imagen) {
        this.imagen = imagen;
    }

    public String getDia() {
        return dia;
    }

    public void setDia(String dia) {
        this.dia = dia;
    }

    public int getDescuento() {
        return descuento;
    }

    public void setDescuento(int descuento) {
        this.descuento = descuento;
    }

    
}

File: Aerolinea/web/js/Vuelo.js
function Vuelo(numero_vuelo,ciudad_origen,ciudad_destino,estado,precio,duracion,hora,oferta,imagen,dia,descuento){
	this.Vuelo(numero_vuelo,ciudad_origen,ciudad_destino,estado,precio,duracion,hora,oferta,imagen,dia,descuento);
}
Vuelo.prototype={
        numero_vuelo:0,
        ciudad_origen:"",
        ciudad_destino:"",
        estado:"",
        precio:"",
        duracion:0,
        hora:0,
        oferta:"",
        imagen:"",
        dia:"",
        descuento:0,
	Vuelo:function(numero_vuelo,ciudad_origen,ciudad_destino,estado,precio,duracion,hora,oferta,imagen,dia,descuento){
		this.numero_vuelo=numero_vuelo;
		this.ciudad_origen=ciudad_origen;
		this.ciudad_destino=ciudad_destino;
		this.estado=estado;
		this.precio=precio;
                this.duracion = duracion;
                this.hora =  hora;
                this.oferta = oferta;
                this.imagen = imagen;
                this.dia = dia;
                this.decuento= descuento;
	},
        toString:function(){
	  return this.numero_vuelo;
	}
}
Vuelo.from= function(plain){
    vuelo = new Vuelo(plain.numero_vuelo,plain.ciudad_origen,plain.ciudad_destino,plain.estado,
    plain.precio,plain.duracion,plain.hora,plain.oferta,plain.imagen,plain.dia,plain.descuento);
    return vuelo;
};

Vuelo.to = function(vuelo){
    return{
        _class : 'Vuelo',
        codigo: vuelo.numero_vuelo,
        ciudad_origen: vuelo.ciudad_origen,
        ciudad_destino: vuelo.ciudad_destino,
        estado: vuelo.estado,
        precio: vuelo.precio,
        duracion: vuelo.duracion,
        hora: vuelo.hora,
        oferta: vuelo.oferta,
        imagen: vuelo.imagen,
        dia: vuelo.dia,
        descuento: vuelo.descuento
    };
};
